scalafix-core/src/main/scala/scalafix/rewrite/ExplicitReturnTypes.scala

```diff
@@ -58,14 +85,39 @@ case class ExplicitReturnTypes(mirror: Mirror)
         typ <- defnType(defn)
       } yield ctx.addRight(replace, s": ${typ.treeSyntax}")
     }.to[Seq]
+
+    import ctx.config.explicitReturnTypes._
+
+    def matchesMemberVisibility(mods: Seq[Mod]): Boolean =
+      memberVisibility.contains(visibility(mods))
+
+    def matchesMemberKind(defn: Defn): Boolean =
+      kind(defn).exists(memberKind.contains)
+
+    def matchesSimpleDefinition(rhs: Term): Boolean =
+      rhs.is[Lit] && skipSimpleDefinitions
+
     tree
       .collect {
         case t @ Defn.Val(mods, _, None, body)
-            if t.hasMod(mod"implicit") &&
-              !isImplicitly(body) =>
+            if t.hasMod(mod"implicit") && !isImplicitly(body)
+              || !matchesSimpleDefinition(body)
```

Comment:
Indeed.
I have extracted it into a helper function, and happen to find that `implicit var` was not include in rewrite scope. I believe it should be, and have applied the same helper function into `Var` as `Val` or `Def`.